src/bgw/job.c

```diff
@@ -246,9 +251,9 @@ ts_bgw_job_get_all(size_t alloc_size, MemoryContext mctx)
 	};
 	ScannerCtx scanctx = {
 		.table = catalog_get_table_id(catalog, BGW_JOB),
-		.index = InvalidOid,
+		.index = catalog_get_index(ts_catalog_get(), BGW_JOB, BGW_JOB_PKEY_IDX),
```

Comment:
could you add a comment on the need to use the index : i.e. we are depending on the sorted order.